paddle/phi/kernels/onednn/conv_transpose_kernel.cc

```diff
@@ -338,46 +346,99 @@ void Execute(const OneDNNContext& dev_ctx,
+
+    std::unordered_map<int, dnnl::memory> args = {
+        {DNNL_ARG_SRC, *src_memory_p},
+        {DNNL_ARG_WEIGHTS, *weights_memory_p},
+        {DNNL_ARG_DST, *dst_memory_p}};
+
+    if (bias) {
+      auto bias_memory_p =
+          std::make_shared<dnnl::memory>(deconvolution_cache->bias_mem);
+      args.insert({DNNL_ARG_BIAS, *bias_memory_p});
+    }
+    auto& astream = OneDNNContext::tls().get_stream();
+    conv_p->execute(astream, args);
+    astream.wait();
+    out->set_mem_desc(dst_memory_p->get_desc());
+  } else {
+    ConvTransposeOneDNNHandlerT<T, float, T_out> handler(dev_ctx,
+                                                         x,
+                                                         filter,
+                                                         bias,
+                                                         strides,
+                                                         paddings,
+                                                         padding_algorithm,
+                                                         groups,
+                                                         dilations,
+                                                         out);
+
+    auto src_memory_p = handler.AcquireSrcMemoryWithReorder(x);
+    // Caching Key for weights is needed
+    std::string key =
+        funcs::CreateKey(dev_ctx,
+                         dev_ctx.GetInputsName("Input")[0],
+                         dev_ctx.GetInputsName("Filter")[0],
+                         (bias ? dev_ctx.GetInputsName("Bias")[0] : ""));
+    key = funcs::ExtendKeyWithThreadInfoIfNeeded(dev_ctx, key);
+    auto weights_memory_p =
+        handler.AcquireWeightsMemoryWithReorder(dev_ctx, key, filter, groups);
+
+    std::shared_ptr<dnnl::memory> dst_memory_p =
+        handler.template AcquireDstMemory<T_out>(out);
+    auto conv_p = handler.AcquireForwardPrimitive();
+
+    std::unordered_map<int, dnnl::memory> args = {
+        {DNNL_ARG_SRC, *src_memory_p},
+        {DNNL_ARG_WEIGHTS, *weights_memory_p},
+        {DNNL_ARG_DST, *dst_memory_p}};
+
+    std::shared_ptr<dnnl::memory> bias_memory_p;
+
+    if (bias) {
+      bias_memory_p = handler.AcquireBiasMemoryWithReorder(dev_ctx, key, bias);
+      args.insert({DNNL_ARG_BIAS, *bias_memory_p});
+    }
+    auto& astream = OneDNNContext::tls().get_stream();
+    conv_p->execute(astream, args);
+    astream.wait();
+    out->set_mem_desc(dst_memory_p->get_desc());
+
+    auto cache = std::make_shared<DeconvolutionCache>();
+    cache->deconvolution_forward = *conv_p;
+    cache->src_mem = *src_memory_p;
+    cache->weights_mem = *weights_memory_p;
+    cache->dst_mem = *dst_memory_p;
+    if (bias) {
+      cache->bias_mem = *bias_memory_p;
+    }
+
+    dev_ctx.SetBlob(cache_key, cache);
   }
-  auto& astream = OneDNNContext::tls().get_stream();
-  conv_p->execute(astream, args);
-  astream.wait();
-  out->set_mem_desc(dst_memory_p->get_desc());
```

Comment:
Seems there are some common code can be extracted outside the condition? Like initialization of primitive and mem and args and execution. (In case you want to reduce the lines lol)